platform/lib/middleware/csp.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const contentSecurityPolicy = require('helmet-csp');
+const config = require('@lib/config');
+
+const getDynamicHosts = () =>
+  ['playground', 'preview', 'go', 'log']
+    .map((key) => {
+      const {[key]: hostConfig} = config.hosts || {};
+      if (hostConfig) {
+        return `${config.getHost(hostConfig)}/`;
+      }
+    })
+    .filter((v) => v);
+
+module.exports = (req, res, next) => {
+  const directives = {
+    defaultSrc: ['*', 'data:', 'blob:'],
+    workerSrc: [`'self'`, 'blob:'],
+    scriptSrc: [
+      'blob:',
+      `'unsafe-inline'`,
+      'https://cdn.ampproject.org/v0.js',
+      'https://cdn.ampproject.org/v0/',
+      'https://cdn.ampproject.org/sw/',
+      'https://cdn.ampproject.org/viewer/',
+      'https://cdn.ampproject.org/rtv/',
+      'https://www.googletagmanager.com/gtag/js',
+      ...getDynamicHosts(),
+    ],
+    objectSrc: [`'none'`],
+    styleSrc: [
+      `'unsafe-inline'`,
+      'https://cdn.ampproject.org/rtv/',
+      ...getDynamicHosts(),
+    ],
+    reportUri: ['https://csp-collector.appspot.com/csp/amp'],
```

Comment:
Shouldn't the [`/csp-report`](https://github.com/ampproject/amp.dev/pull/5072/commits/2a3158a2bbb031e7ac13254edc826940605c9531#diff-2b890bf0ec18da7e4c03e8bf4568a4c7c850411978807b0980a9338fd64a0050R27) uri be used here?

Reply:
🤦‍♂️

Reply:
fixed